### PR TITLE
Add tasks planner

### PR DESCRIPTION
**Pull request by andreibancioiu**
Added `TasksPlanner` component (along with unit and integration tests).
 
This component is responsible with splitting the **long ETL task** of _copying Elastic Search indexes_ (from a public MultiversX indexer instance) into **many, smaller ETL tasks**. The tasks produced by `TasksPlanner` are saved in a Google Firestore collection, to be taken and resolved / completed at a later time (by separate worker processes).

### CODE REVIEW

File: multiversxetl/cli/plan_tasks.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        start_timestamp,
+        end_timestamp,
+        granularity
+    )
+
+    storage.add_tasks(new_tasks)
+
+
+@click.command(context_settings=dict(help_option_names=["-h", "--help"]))
+@click.option("--indexer-url", type=str, help="The indexer URL (Elastic Search instance).")
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply multiversxetl/cli/plan_tasks.py:64]: Fixed.

File: multiversxetl/planner/tasks_planner.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def plan_tasks_with_intervals(
+            self,
+            indexer_url: str,
+            bq_dataset: str,
+            initial_start_timestamp: int,
+            initial_end_timestamp: int,
+            granularity_seconds: int
+    ) -> List[Task]:
+        """
+        Plans ETL tasks for Elastic Search indexes that support time intervals.
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply multiversxetl/planner/tasks_planner.py:22]: Fixed.

**Review by bogdan-rosianu — APPROVED** [review]

File: multiversxetl/cli/plan_tasks.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+from multiversxetl.constants import SECONDS_IN_DAY, SECONDS_IN_MINUTE
+from multiversxetl.planner import (TasksPlanner, TasksWithIntervalStorage,
+                                   TasksWithoutIntervalStorage)
+from multiversxetl.planner.tasks import count_tasks_by_status
+
+logging.basicConfig(level=logging.INFO)
+
+
+@click.command(context_settings=dict(help_option_names=["-h", "--help"]))
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply multiversxetl/cli/plan_tasks.py:15]: Will be done in a separate PR.

File: multiversxetl/cli/plan_tasks.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+@click.option("--gcp-project-id", type=str, help="The GCP project ID.")
+@click.option("--bq-dataset", type=str, required=True, help="The BigQuery dataset (destination).")
+@click.option("--start-timestamp", type=int, help="The start timestamp (e.g. genesis time).")
+@click.option("--end-timestamp", type=int, help="The end timestamp (e.g. a recent one).")
+@click.option("--granularity", type=int, default=SECONDS_IN_DAY, help="Task granularity, in seconds.")
+def plan_tasks_with_intervals(indexer_url: str, gcp_project_id: str, bq_dataset: str, start_timestamp: int, end_timestamp: int, granularity: int):
+    storage = TasksWithIntervalStorage(gcp_project_id)
+    planner = TasksPlanner()
+
+    if not end_timestamp:
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply multiversxetl/cli/plan_tasks.py:48]: Will be done in a separate PR.

File: multiversxetl/planner/tasks_planner.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        tasks: List[Task] = []
+
+        for index_name in INDEXES_WITHOUT_INTERVALS:
+            id = self._next_task_id()
+            task = Task(id, indexer_url, index_name, bq_dataset)
+            tasks.append(task)
+
+        return tasks
+
+    def _next_task_id(self, ) -> str:
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply multiversxetl/planner/tasks_planner.py:51]: Will be done in a separate PR.

**Pull request merged**